pkg/cortex/cortex.go

```diff
@@ -59,22 +60,24 @@ type Config struct {
 	PrintConfig bool       `yaml:"-"`
 	HTTPPrefix  string     `yaml:"http_prefix"`
 
-	Server         server.Config            `yaml:"server,omitempty"`
-	Distributor    distributor.Config       `yaml:"distributor,omitempty"`
-	Querier        querier.Config           `yaml:"querier,omitempty"`
-	IngesterClient client.Config            `yaml:"ingester_client,omitempty"`
-	Ingester       ingester.Config          `yaml:"ingester,omitempty"`
-	Storage        storage.Config           `yaml:"storage,omitempty"`
-	ChunkStore     chunk.StoreConfig        `yaml:"chunk_store,omitempty"`
-	Schema         chunk.SchemaConfig       `yaml:"schema,omitempty" doc:"hidden"` // Doc generation tool doesn't support it because part of the SchemaConfig doesn't support CLI flags (needs manual documentation)
-	LimitsConfig   validation.Limits        `yaml:"limits,omitempty"`
-	Prealloc       client.PreallocConfig    `yaml:"prealloc,omitempty" doc:"hidden"`
-	Worker         frontend.WorkerConfig    `yaml:"frontend_worker,omitempty"`
-	Frontend       frontend.Config          `yaml:"frontend,omitempty"`
-	QueryRange     queryrange.Config        `yaml:"query_range,omitempty"`
-	TableManager   chunk.TableManagerConfig `yaml:"table_manager,omitempty"`
-	Encoding       encoding.Config          `yaml:"-"` // No yaml for this, it only works with flags.
-	TSDB           tsdb.Config              `yaml:"tsdb" doc:"hidden"`
+	Server           server.Config            `yaml:"server,omitempty"`
+	Distributor      distributor.Config       `yaml:"distributor,omitempty"`
+	Querier          querier.Config           `yaml:"querier,omitempty"`
+	IngesterClient   client.Config            `yaml:"ingester_client,omitempty"`
+	Ingester         ingester.Config          `yaml:"ingester,omitempty"`
+	Storage          storage.Config           `yaml:"storage,omitempty"`
+	ChunkStore       chunk.StoreConfig        `yaml:"chunk_store,omitempty"`
+	Schema           chunk.SchemaConfig       `yaml:"schema,omitempty" doc:"hidden"` // Doc generation tool doesn't support it because part of the SchemaConfig doesn't support CLI flags (needs manual documentation)
+	LimitsConfig     validation.Limits        `yaml:"limits,omitempty"`
+	Prealloc         client.PreallocConfig    `yaml:"prealloc,omitempty" doc:"hidden"`
+	Worker           frontend.WorkerConfig    `yaml:"frontend_worker,omitempty"`
+	Frontend         frontend.Config          `yaml:"frontend,omitempty"`
+	QueryRange       queryrange.Config        `yaml:"query_range,omitempty"`
+	TableManager     chunk.TableManagerConfig `yaml:"table_manager,omitempty"`
+	Encoding         encoding.Config          `yaml:"-"` // No yaml for this, it only works with flags.
+	TSDB             tsdb.Config              `yaml:"tsdb" doc:"hidden"`
+	DeletesConfig    chunk.DeleteStoreConfig  `yaml:"deletes,omitempty"`
```

Comment:
Can we find a better name for the YAML config? `deletes` looks a bit unclear to me, if I look at it with the eyes of the first time user.

Also, is it necessary to have this config at root level? Is there a better place inside any other root config block we already have? _Thinking loudly_